pepper-apis/src/main/java/org/broadinstitute/ddp/Housekeeping.java

```diff
@@ -511,6 +512,32 @@ private static void respondToGAEStartHook(String envPort) {
             response.status(200);
             return "";
         });
+        Spark.post("/heapdump", (request, response) -> {
+            String postedPassword = request.queryParams("password");
+            if (StringUtils.isNotEmpty(postedPassword)) {
+                String configuredPwd = ConfigManager.getInstance().getConfig().getString(ConfigFile.HEALTHCHECK_PASSWORD);
+                if (!configuredPwd.equals(postedPassword)) {
+                    LOG.error("Invalid password for heapdump call");
+                    response.status(404);
+                    return "";
+                }
+            } else {
+                LOG.error("Missing password for heapdump call");
+                response.status(404);
+                return "";
+            }
+            try {
+                String projectId = ConfigManager.getInstance().getConfig().getString(ConfigFile.GOOGLE_PROJECT_ID);
+                String bucketName = projectId + "-heap-dumps";
+                new JavaHeapDumper().dumpHeap(projectId, bucketName);
```

Comment:
Each env will get their corresponding bucket. Did not want to mess with permissions to allow prod to save to dev buckets or vice-versa. Only dev bucket exists at this time.